### PR TITLE
Fix uppercase computer name issue

### PR DESCRIPTION
**Pull request by twangboy**
Fixes Capitalized Computer name for states and for current in set_computer_name function
No way to fix the capitalization in Pending Computer name. (That's how it's stored in the registry). I couldn't find where windows stores the un-capitalized version of the pending computer name.

Execution Module returns:

```
gringo:
    ----------
    Computer Name:
        ----------
        Current:
            Gringo
        Pending:
            GRINGO_TEST
```

State returns:

```
gringo:
----------
          ID: set_name
    Function: system.computer_name
        Name: Gringo-Test
      Result: True
     Comment: Computer name successfully set to 'Gringo-Test' (reboot required for change to take effect)
     Started: 16:28:24.955000
    Duration: 16.0 ms
     Changes:
              ----------
              new:
                  Gringo-Test
              old:
                  Gringo

Summary for gringo
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
```

Also adds more computer information for get_system_info.

Fixes: https://github.com/saltstack/salt/issues/31290
Supercedes: https://github.com/saltstack/salt/pull/31222

### CODE REVIEW

**Comment by fredrikaverpil** [comment]
Great job. Thank you!

**Comment by cachedout** [comment]
@twangboy This does not pass a Windows test. Please take another look.

**Comment by twangboy** [comment]
@cachedout Justin's going to help me figure the tests out.

**Comment by twangboy** [comment]
@cachedout Looks like it's fixed.

File: salt/states/win_system.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def computer_name(name):
     if result is not False:
         after_name = result['Computer Name']['Current']
         after_pending = result['Computer Name'].get('Pending')
-        if ((after_pending is not None and after_pending == name) or
+        if ((after_pending is not None and after_pending == name.upper()) or
                 (after_pending is None and after_name == name)):
```
Comment (cachedout) [inline salt/states/win_system.py]: You removed the `.upper()` for `name` in L91 and put it back in on L118 but not in the comparison in L119. Is this a bug?

**Comment by cachedout** [comment]
I think I understand the case for this, but I'm not convinced this should go into a point release. What do you think about moving this to develop instead? Also, has this been testing between versions to make sure that name is not reset just by virtue of upgrading Salt?

File: salt/states/win_system.py (hunk 1 above)
Comment (twangboy) [inline salt/states/win_system.py]: @cachedout I want the computer name to actually be the case the user passed and not capitalize it. The `system.get_pending_computer_name` function always returns a capitalized host name, so you must capitalize `name` when you make the comparison. I could not find another way to get a not-all-caps pending name in windows.

**Comment by twangboy** [comment]
@cachedout I think the real bug here was that if you set a computer name using the execution module the case would match your new computer name. If you set the computer name using a state file, it would convert your computer name to all caps.

I'm not sure what you mean by your last sentence about the name being reset by virtue of upgrading salt. Upgrading salt shouldn't change the host name of a machine.

**Comment by cachedout** [comment]
I don't see that as a bug, per se. I just want to make sure that the same state, applied before this version and again after does not result in the capitalization of the hostname on the machine itself being affected.

**Comment by twangboy** [comment]
@cachedout Oh, I see. States previous to this fix set the host name ALL CAPS no matter the capitalization in the state file. After this fix, they will maintain the capitalization as it is in the state file. So, yes, it will rename the machine to honor the capitalization in the state file.

**Comment by cachedout** [comment]
In that case, I _really_ don't want this in a point release. This should go in develop, along with an entry in the release notes indicating the potential for changes in the capitalization of computer names.

**Comment by rallytime** [comment]
Would it also make sense to follow a deprecation path and add some deprecation warnings to the change in behavior?

**Comment by cachedout** [comment]
I feel like a deprecation path is probably overkill here, but I certainly wouldn't object to one.

**Pull request closed without merging**